### DISCUSSION TITLE
Overset Grid Support

### DISCUSSION THREAD

**Discussion by kursatyurt** [su2code/SU2#1906] [Q&A]
**Is your feature request related to a problem? Please describe.**

Overset grids are very advantageous for storage separation or rotorcraft dynamic problems where rigid bodies move independently.

In the past, it is shown that SU2 can be adapted to work with the overset grid as shown in [video](https://www.youtube.com/watch?v=tsr-1Qyk1AQ) [paper](https://arc.aiaa.org/doi/abs/10.2514/6.2015-3428) however this future newer added to community version. 

**Describe the solution you'd like**

I would like to get the idea of the community about overset grids, the possible implementation, and the work effort required for such a modification. 

Any contribution/suggestion is highly appreciated.

**Comment by bigfooted** [comment]
An active group working on similar topics is Alberto Guardone's group, here is a recent paper:
https://www.researchgate.net/publication/353570467_Development_and_Preliminary_Assessment_of_the_Open-Source_CFD_toolkit_SU2_for_Rotorcraft_Flows

Is there a specific problem that you would like to tackle in rotorcraft dynamics?

**Reply by kursatyurt** [reply]
AFAIK, they are using rotating frame and use mesh deformation for the giving control angles and there is at least no publicly available code base.

It is completely legit, however I think overset mesh also open more flexible meshing options.

**Reply by bigfooted** [reply]
My question then is if you want to solve rotorcraft dynamics problems, or you want to work specifically on overset meshes? 
It can be that overset meshes are the best solution for whatever you have in mind, but if there is currently no code available (and code from 2015 should probably be considered 'unavailable') then you either have to start implementing it from scratch, or look around for whatever *is* available that can solve your problem. In any case, the sliding mesh implementation probably contains a lot of functionality that you could reuse for overset meshes so it would be a good place to start imo.

In any case it would be interesting to have. The work effort: it depends on your initial knowledge. For you it's less time than for a random engineering student. But still, I'd say in academic units it's at least an MSc. thesis project.

**Reply by kursatyurt** [reply]
Hi, sorry for the late reply. I am looking for rotorcraft simulation specifically. I had to discuss this topic with an experienced person in rotorcraft simulation. I also tried to reach people who implemented overset grid to SU2 and Myles Morelli from PoliMilano but did not get an answer yet.

I also looked around the open-source projects for overset grid assembly/interpolation part but it seems there is no proper tool available in this area at least open-source side. 

Using a multi-block grid and surface deformation + rotating frame approach seems to be the easiest way to just handle the problem. However, it is best practice in rotorcraft simulation to  have separate background and near-field meshes which can be easily changed (cartesian background to cylindrical/spherical background mesh, etc.)

Probably I could get enough manpower to implement an overset grid approach to SU2. ATM, I am not sure which components are required to write for the overset library or available (Assembly of Mesh, Hole Cutting, Interpolation what else). Currently, I am looking for http://overtureframework.org/ as it seems the only well-documented option to use.

**Reply by bigfooted** [reply]
Maybe also check what Joaquim Martins uses for his cfd/adjoint solvers, ADFlow can do overset meshes:
https://mdolab.engin.umich.edu/
And of course check what openfoam does.

**Comment by kursatyurt** [comment]
Dear @bigfooted, sorry for disturbing you again, today I reached out to Alberto Guardone, and he told me the code should be available in the SU2 code base however I couldn't find the PR or any topic about it. Do you have any idea if it was included in SU2 main repo and removed in some restructuring?

**Reply by bigfooted** [reply]
What's in develop is everything related to "Sliding State" in the code, so CSlidingInterface.cpp, the regression tests for "Sliding Mesh" and the testcases in TestCases/sliding_interface.  I guess this is what Alberto is referring to.

**Reply by kursatyurt** [reply]
Probably yes. However, I am actually looking for the implementation partly shown in https://su2foundation.org/wp-content/uploads/2019/05/Morelli_SU2_Dev4.pdf. Specifically `GRID_MOVEMENT_KIND= ROTORCRAFT`.

In the presentation it mentioned `Once at a stage where there is complete verification of the method and validation of the
results there will be a pull request to merge the feature_ROTORCRAFT branch into the
main release branch.` but `feature_ROTORCRAFT` is missing at all

**Reply by tbellosta** [reply]
Hi, 
I am one of the authors of the paper @bigfooted linked a few messages above this one.  
As you were saying the approach made use of either the rotating frame (for hover conditions) or the sliding meshes (for forward flight) together with mesh deformation to comply with the blades kinematics. The only addition to the code was the prescription of the surface movement that represented the rigid blades pitch, flap  and lead-lag. Myles added a rigid blade kinematics as a possible surface movement type (i.e. the option ROTORCRAFT could be selected as a possible SURFACE_MOVEMENT= in the SU2 config) which prescribed a time dependent displacement for the blades. You can see a snapshot of the code in slide 8 of the presentation you linked.
I'm not sure the code is available on any public branch of the github repo. Also, as far as I know, the work Myles presented at the dev Meeting in 2019 was done on version 6 of the code, and the surface/volume movement logic changed a bit since then. I can try contact him and see if we can bring the code up to date and make a pull request, bu can't promise much.

**Reply by kursatyurt** [reply]
> Hi, I am one of the authors of the paper @bigfooted linked a few messages above this one. As you were saying the approach made use of either the rotating frame (for hover conditions) or the sliding meshes (for forward flight) together with mesh deformation to comply with the blades kinematics. The only addition to the code was the prescription of the surface movement that represented the rigid blades pitch, flap and lead-lag. Myles added a rigid blade kinematics as a possible surface movement type (i.e. the option ROTORCRAFT could be selected as a possible SURFACE_MOVEMENT= in the SU2 config) which prescribed a time dependent displacement for the blades. You can see a snapshot of the code in slide 8 of the presentation you linked. I'm not sure the code is available on any public branch of the github repo. Also, as far as I know, the work Myles presented at the dev Meeting in 2019 was done on version 6 of the code, and the surface/volume movement logic changed a bit since then. I can try contact him and see if we can bring the code up to date and make a pull request, bu can't promise much.

Thank you very much for contact. Even if you don't have time to refactor it, an older version would be useful for me, I can refactor it for current version. 

A tested logic is a good start point. 

Again thank you very much.

**Reply by kursatyurt** [reply]
@tbellosta is there any news?

**Reply by tbellosta** [reply]
Yes, here you go:
https://github.com/mcmorelli/SU2/tree/feature_rotorcraftMovement

It is updated to version 7.0.3 of the code.  All options related to the rotorcraft surface movement are listed in the config template. The work is of Myles Morelli (@mcmorelli)

**Reply by kursatyurt** [reply]
Thanks!!.

**Reply by stevosg99** [reply]
Has this been integrated into v8.0.1?

**Reply by pcarruscag** [reply]
no
